### PR TITLE
feat: add support for linux aarch64

### PR DESCRIPTION
**Pull request by Kavinjsir**
Resolve:
https://rilldata.slack.com/archives/C01190F1R1D/p1724443548781379

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -64,6 +64,25 @@ builds:
     ldflags:
       - -s -w -X main.Version={{.Version}} -X main.Commit={{.ShortCommit}} -X main.BuildDate={{.Date}}
 
+  - id: linux_aarch64
+    main: ./cli/main.go
+    binary: rill
+    goos:
+      - linux
+    goarch:
+      - arm64
```
Comment (Kavinjsir) [inline .goreleaser.yaml:73]: Looks like go env didn't provide literal arch name as `aarch`, so I'd assume `arm64` might be fit in this case?

**Comment by himadrisingh** [comment]
can we try running a nightly/snapshot from this branch and test it out?

File: .goreleaser.yaml (hunk 1 above)
Comment (begelundmuller) [reply .goreleaser.yaml:73]: Yeah, `aarch64` refers to the same architecture as `arm64`

**Comment by begelundmuller** [comment]
Here's a PR on top of this PR that adds DuckDB extension embedding for the new build target: https://github.com/rilldata/rill/pull/5519.

File: scripts/install.sh [hunk 2]
```diff
@@ -10,6 +10,8 @@ initPlatform() {
         PLATFORM="darwin_amd64"
     elif [ "$OS" = "linux" ] && [ "$ARCH" = "x86_64" ]; then
         PLATFORM="linux_amd64"
+    elif [ "$OS" = "linux" ] && [ "$ARCH" = "arm64" ]; then
```
Comment (kaspersjo) [inline scripts/install.sh:13]: I think on a lot of modern arm64 linux system `ARCH` will resolve to `aarch64` not `arm64`, so we might need to add support for both cases here

**Review by kaspersjo — CHANGES_REQUESTED** [review]

**Comment by Kavinjsir** [comment]
> can we try running a nightly/snapshot from this branch and test it out?

Sounds good! Schedule an action for a nightly release:
https://github.com/rilldata/rill/actions/runs/10569133157

File: scripts/install.sh (hunk 2 above)
Comment (Kavinjsir) [reply scripts/install.sh]: Good catch! Just updated the code to:
```bash
    elif [ "$OS" = "linux" ] && { [ "$ARCH" = "arm64" ] || [ "$ARCH" = "aarch64" ]; }; then
        PLATFORM="linux_arm64"
```

**Review by himadrisingh — APPROVED** [review]

**Review by himadrisingh — APPROVED** [review]

**Review by kaspersjo — APPROVED** [review]

**Pull request merged**